### ISSUE TITLE
Ingress support in ECK Helm charts

### ISSUE THREAD

**Issue by gbanasiak** [elastic/cloud-on-k8s#6285]
As Elastic maintenance of non-ECK Helm charts is drawing to an end (see https://github.com/elastic/helm-charts/issues/1731) there's a group of customers who will migrate from non-ECK to ECK Helm charts. Today there's no full feature parity between non-ECK and ECK Helm charts and one of the missing components is Ingress support.

Non-ECK Helm charts include Ingress templates for APM server, Elasticsearch, Kibana and Logstash. Leaving Logstash aside (see https://github.com/elastic/cloud-on-k8s/issues/1453) a reasonable starting point for Ingress support in ECK Helm charts would be feature parity with what was offered in [[1]](https://github.com/elastic/helm-charts/blob/main/apm-server/templates/ingress.yaml), [[2]](https://github.com/elastic/helm-charts/blob/main/elasticsearch/templates/ingress.yaml), and [[3]](https://github.com/elastic/helm-charts/blob/main/kibana/templates/ingress.yaml).

Let's use Ingress case to decide whether ECK Helm charts are intended _solely_ as Helm "wrappers" for ECK CRDs (which is the current state). If yes, then Helm users would need to rely on custom charts or charts provided by 3rd parties to manage resources such as Ingress.

**Comment by rkubik-hostersi** [comment]
Good point, however the fact, that ECK components charts are Enterprise feature, I suppose, that a 3rd party/community charts will be born soon to allow full functionality and be free.
In my opinion, licensing the charts this way is a mindless move...

**Comment by naemono** [comment]
Fyi: https://github.com/elastic/cloud-on-k8s/pull/7962 the 2.14 release will include support for Elasticsearch and Kibana ingress support in the Helm charts. Other products will come in subsequent releases.

Also, the Helm charts are no longer Enterprise only as of [v2.12](https://github.com/elastic/cloud-on-k8s/releases/tag/v2.12.0) @rkubik-hostersi

**Comment by barkbay** [comment]
>  Other products will come in subsequent releases.

I have removed the `v2.14.0` labels as we are now in feature freeze for that release. Feel free to target the next one if you think it makes sense. Thanks!

**Comment by naemono** [comment]
Closing in favor of #9085
